### PR TITLE
6205 show file in edit media

### PR DESCRIPTION
**Pull request by emitche**
- In edit media modal, display the currently attached file

### CODE REVIEW

**Comment by emitche** [comment]
These changes make use of the existing way media is displayed. The full file name is shown in the edit media modal. The link opens the file in a new window.

There are no actions on hover (edit, delete) as in the media list.

image

File: app/helpers/media_helper.rb [hunk 1]
```diff
@@ -10,13 +10,20 @@ def media_thumbnail(media_item)
     end
   end
 
-  def media_title(media_item)
-    ext = File.extname(media_item.item.file.path)
-    file_name = truncate(File.basename(media_item.item.file.path, ext), length: 12)
+  def media_title(media_item, shorten: true)
```
Comment (hooverlunch) [inline app/helpers/media_helper.rb]: Note that I changed this to a named parameter, which makes it easier to read below.

File: app/helpers/media_helper.rb [hunk 2]
```diff
@@ -10,13 +10,20 @@ def media_thumbnail(media_item)
...
+      ext = File.extname(media_item.item.file.path)
+      file_name = truncate(File.basename(media_item.item.file.path, ext), length: 12)
+      content_tag(:div, class: "media-title", title: full_name) do
+        concat(content_tag(:span, file_name))
+        concat(content_tag(:div, ext.downcase))
+      end
+    else
+      content_tag(:div, class: "media-title") do
+        concat(content_tag(:span, full_name))
+      end
```
Comment (hooverlunch) [inline app/helpers/media_helper.rb]: Nice code here otherwise.

File: app/views/admin/media/_modal_content.html.slim [hunk 3]
```diff
@@ -3,10 +3,18 @@ div.modal-header
...
-div.modal-body
+div.modal-body.media-modal
   = simple_form_for [:admin, @media], url: @submit_url, remote: true do |f|
     = error_notification(f)
+
+    - unless @media.new_record?
+      div.media-item data-media-id="#{@media.id}"
+        .thumb
+          = link_to media_thumbnail(@media), @media.item.url, target: "_blank"
+          = media_title(@media, shorten: false)
```
Comment (hooverlunch) [inline app/views/admin/media/_modal_content.html.slim]: Here we see `shorten: false` so it's easier to see what the `false` means.

**Comment by hooverlunch** [comment]
Oops reviewed this by mistake. But comments/changes are still valid :)

**Comment by emitche** [comment]
Thanks for the review, @hooverlunch. I need to make some UX tweaks. I'll also make these code changes and resubmit.

File: app/views/admin/media/_modal_content.html.slim (hunk 3 above)
Comment (emitche) [inline app/views/admin/media/_modal_content.html.slim]: Makes sense. Thanks.

**Comment by emitche** [comment]
@hooverlunch This has not changed a whole lot since you last looked at it. I added a few lines of JavaScript and merged in develop. Thanks.

File: app/assets/stylesheets/admin/media/_media.scss [hunk 4]
```diff
@@ -18,6 +18,31 @@ $media-block-width: 100px;
...
+      font-size: 1.4em;
+    }
+  }
+}
+
+.media-browser {
+  h4 {
+    font-size: 15px;
+  }
+
```
Comment (hooverlunch) [inline app/assets/stylesheets/admin/media/_media.scss]: Nice refactors.

**Comment by emitche** [comment]
Thanks a bunch, @hooverlunch!

**Pull request merged**